_source/_posts/2022-01-04-angular-auth0-quickly.md

```diff
@@ -0,0 +1,177 @@
+
+1. React
+2. ASP.NET MVC
+3. Angular
+4. Ruby on Rails
+5. AngularJS
+6. Vue.js
+
+Angular is still very popular among developers. If you're developing an Angular app today, you probably need a way to authenticate your users. That's where OpenID Connect (OIDC) can help you. OIDC is a layer on top of OAuth 2.0 that provides identity. 
+
+In this tutorial, I'll how you to add OIDC authentication with Auth0 to a new Angular app in just a few steps. 
+
+**Prerequisites:**
+
+* A favorite text editor or IDE. I recommend [IntelliJ IDEA](https://www.jetbrains.com/idea/), but I know many JavaScript developers prefer [Visual Studio Code](https://code.visualstudio.com/).
+* [Node.js](http://nodejs.org/) v14+ and npm installed.
+
+{% include toc.md %}
+
+## Create an Angular app
+
+First, you'll need to create a new Angular app with routing enabled.
+
+```shell
+npx @angular/cli@13 new auth0-demo --routing
+```
+
+Select your favorite stylesheet format. The selection you make doesn't matter for this example.
+
+## Add Auth with OpenID Connect
+
+To add authentication with Auth0, you'll first need a [free Auth0 account](https://auth0.com/signup). Install the [Auth0 CLI](https://github.com/auth0/auth0-cli#installation) and run `auth0 login` to register your account. Then, run `auth0 apps create`. Specify a name and description of your choosing. Select **Single Page Web Application** and use `http://localhost:4200/home` for the Callback URL. Specify `http://localhost:4200` for the rest of the URLs.
+
+{% img blog/angular-auth0/auth0-apps-create.png alt:"auth0 apps create" width:"800" %}{: .center-image }
+
+Once you have a new Angular app and Auth0 OIDC set up, you can use [OktaDev Schematics](https://github.com/oktadev/schematics#angular--auth0) to add OAuth 2.0 and OIDC support to your Angular app.
+
+```shell
+ng add @oktadev/schematics --auth0
+```
+
+You'll be prompted for an issuer and client ID. You should have these from the OIDC app you just created. The issuer will be displayed right above your client ID.
+
+```shell
+=== dev-0ua1y-go.us.auth0.com application created
+
+  CLIENT ID            TJgZxGnlSkqUe6JfMFSipZcFsbpl6LS2
+```
+
+This process will perform the following steps for you:
+
+1. Install the [Auth0 Angular SDK](https://github.com/auth0/auth0-angular).
+2. Add `src/app/auth-routing.module.ts` with your OIDC configuration and initialization logic.
+3. Configure an `AuthHttpInterceptor` that adds an `Authorization` header with an access token to outbound requests.
+4. Create a `HomeComponent` and configure it with authentication logic.
+5. Update unit tests for `AppComponent` and `HomeComponent` to mock Auth0.
+
+To see all the changes it makes to your files, see [this pull request on GitHub](https://github.com/oktadev/auth0-angular-example/pull/1/files). 
+
+### Use the Auth0 Console instead of the CLI
```

Comment:
This should be moved up adjacent to the CLI instructions, and reorganized to show that you can accomplish the same goal either way.

Reply:
I tend to de-emphasize using a console over a CLI. This is because the Okta Admin console changed in March 2020 and made 400+ blog posts invalid. I spent months updating them and I'm still scared of future changes that might invalid posts again. To help prevent this in the future, we now only include CLI instructions and link to console instructions. For examples, see https://developer.okta.com/blog/setup.

Reply:
I think focusing on and prioritizing the CLI is 100% the way to go. After re-looking at it I think this location does make sense as it's after the conclusion of all the CLI instructions, it's just a bit strange & stuck out to me on first reading to have a middle step at the end.

Could we perhaps even eliminate this whole section and add just a sentence about the other option for registering an app via the console? I'll resolve this and leave it up to you.